### PR TITLE
Fixes #4568 - explain PassThru better

### LINKED ISSUE #4468: PassThru parameter does not function as described per documentation
Running PS Version: 5.1.17763.134.  -PassThru parameter is not respected as described here.  Even with PassThru specified the original objects supplied are wrapped by to include the additional column.  
In same version if I omit the -passthru parameter I DO achieve the desired results

```
\> $CurrentJobList[0]
Name         Code
----             ----
19xx-Job1   19xx

\> $MisMatches = Compare-Object $jobList18Jun $CurrentJobList -Property name -PassThru
\> $CurrentJobList[0]
SideIndicator  Name       Code
-------------  ----       ----
=>             19xx-Job1  19xx

```
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 18184766-c3b5-3f82-b4d0-8175a93cb3e3
* Version Independent ID: 85f1e969-8216-bed4-2204-409cbd9befd4
* Content: [Compare-Object (Microsoft.PowerShell.Utility)](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/compare-object?view=powershell-5.1)
* Content Source: [reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md)
* Product: **powershell**
* Technology: **powershell-cmdlets**
* GitHub Login: @JamesWTruher
* Microsoft Alias: **jimtru**

### LINKED ISSUE #4568: "Property" output is similar as if "Passthru" is specified
Not sure if this is a bug or expected behavior.

When I use Property to handle comparison, the output is similar as if "Passthru" was specified, but just shows the properties I'm comparing against

Source object:
```
PS C:\WINDOWS\system32> $MonitorSetup_Current


BitsPerPixel : 32
Bounds       : {X=0,Y=0,Width=3440,Height=1440}
DeviceName   : \\.\DISPLAY1
Primary      : True
WorkingArea  : {X=0,Y=0,Width=0,Height=0}

BitsPerPixel : 32
Bounds       : {X=3440,Y=-307,Width=1200,Height=1920}
DeviceName   : \\.\DISPLAY2
Primary      : False
WorkingArea  : {X=0,Y=0,Width=0,Height=0}
```

Comparison object:
```
PS C:\WINDOWS\system32> $MonitorSetup_New


BitsPerPixel : 32
Bounds       : {X=0,Y=0,Width=3440,Height=1440}
DeviceName   : \\.\DISPLAY1
Primary      : True
WorkingArea  : {X=0,Y=0,Width=3440,Height=1400}
```

Property output (comparing DeviceName,Bounds,Primary)
```
PS C:\WINDOWS\system32> Compare-Object -ReferenceObject $MonitorSetup_Current -DifferenceObject $MonitorSetup_New -Property DeviceName,Bounds,Primary

DeviceName   Bounds                                 Primary SideIndicator
----------   ------                                 ------- -------------
\\.\DISPLAY2 {X=3440,Y=-307,Width=1200,Height=1920}   False <=           
```

Full output, no parameters:
```
PS C:\WINDOWS\system32> Compare-Object -ReferenceObject $MonitorSetup_Current -DifferenceObject $MonitorSetup_New

InputObject                                                                                                                       SideIndicator
-----------                                                                                                                       -------------
Screen[Bounds={X=0,Y=0,Width=3440,Height=1440} WorkingArea={X=0,Y=0,Width=3440,Height=1400} Primary=True DeviceName=\\.\DISPLAY1  =>           
Screen[Bounds={X=0,Y=0,Width=3440,Height=1440} WorkingArea={X=0,Y=0,Width=0,Height=0} Primary=True DeviceName=\\.\DISPLAY1        <=           
Screen[Bounds={X=3440,Y=-307,Width=1200,Height=1920} WorkingArea={X=0,Y=0,Width=0,Height=0} Primary=False DeviceName=\\.\DISPLAY2 <=           
```

PSVersionTable output:
```
PS C:\WINDOWS\system32> $PSVersionTable

Name                           Value                                                                                                                            
----                           -----                                                                                                                            
PSVersion                      5.1.17134.858                                                                                                                    
PSEdition                      Desktop                                                                                                                          
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                          
BuildVersion                   10.0.17134.858                                                                                                                   
CLRVersion                     4.0.30319.42000                                                                                                                  
WSManStackVersion              3.0                                                                                                                              
PSRemotingProtocolVersion      2.3                                                                                                                              
SerializationVersion           1.1.0.1                                                                                                                          
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a7615306-98e3-0b79-ac87-58bb191b4349
* Version Independent ID: 85f1e969-8216-bed4-2204-409cbd9befd4
* Content: [Compare-Object (Microsoft.PowerShell.Utility)](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/compare-object?view=powershell-6#feedback)
* Content Source: [reference/6/Microsoft.PowerShell.Utility/Compare-Object.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Utility/Compare-Object.md)
* Product: **powershell**
* Technology: **powershell-cmdlets**
* GitHub Login: @JamesWTruher
* Microsoft Alias: **jimtru**

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

- Fixes #4568 - explain PassThru better
- Fixes #4468 
- Fixes [AB#1637064](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1637064)

## PR Context

Select the type(s) of documents being changed.

**Cmdlet reference & about_ topics**
- [x] Version 7 content
- [x] Version 6 content
- [x] Version 5.1 content

**Conceptual articles**
- [ ] Fundamental conceptual articles
- [ ] Script sample articles
- [ ] DSC articles
- [ ] Gallery articles
- [ ] JEA articles
- [ ] WMF articles
- [ ] SDK articles

## PR Checklist

- [x] I have read the [contributors guide](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/CONTRIBUTING.md) and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords](https://help.github.com/en/articles/closing-issues-using-keywords).
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the
    title and remove the prefix when the PR is ready.

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 1]
```diff
@@ -129,17 +132,87 @@ InputObject                            SideIndicator
...
+**notepad.exe**, from the `$Processes_After` object.
+
+### Example 5: Show the difference when using the PassThru parameter
+
+Normally, `Compare-Object` returns a **PSCustomObject** type with the following properties:
+
+- The **InputObject** being compared
+- The **SideIndicator** property showing which input object the output belongs to
+
+When you use the **PassThru** parameter **Type** of the object is not changed but the instance of
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: Comma after parameter.

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 2]
```diff
@@ -129,17 +132,87 @@ InputObject                            SideIndicator
...
+ToUInt64      Method       ulong IConvertible.ToUInt64(System.IFormatProvider provider)
+TryFormat     Method       bool TryFormat(System.Span[char] destination, [ref] int charsWritten)
+SideIndicator NoteProperty string SideIndicator===
+```
 
-The `Compare-Object` compare the two sets of process objects stored in the `$Processes_Before` and
-`$Processes_After` variables. The output displays the difference, **notepad.exe**, from the
-`$Processes_After` object.
+When using **PassThru**, the original object type (**System.Boolean**) is returned. Note how the
+output display by the default format **System.Boolean** objects didn't display the **SideIndicator**
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: output displayed

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 3]
```diff
@@ -316,17 +389,25 @@ You can send an object down the pipeline to the **DifferenceObject** parameter.
...
 
 ### System.Management.Automation.PSCustomObject
 
 If the objects are different, `Compare-Object` wraps the differing objects in a `PSCustomObject`
-wrapper with a **SideIndicator** property to reference the differences. When you use the
-**PassThru** parameter, `Compare-Object` omits the `PSCustomObject` wrapper around the compared
-objects and returns the differing objects, unchanged.
+wrapper with a **SideIndicator** property to reference the differences.
+
+When you use the **PassThru** parameter **Type** of the object is not changed but the instance of
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: Comma after parameter

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md [hunk 4]
```diff
@@ -316,17 +389,25 @@ You can send an object down the pipeline to the **DifferenceObject** parameter.
...
+the object returned has an added **NoteProperty** named **SideIndicator**. **SideIndicator** shows
+which input object the output belongs to.
 
 ## NOTES
 
+When using the **PassThru** parameter, the output displayed in the console may not include the
+**SideIndicator** property. The default format view of the for the object type output by
+`Compare-Object` does not include the **SideIndicator** property. For more information see Example 5
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: The see Example 5 should be a link to example 5.

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 1 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 3 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md (hunk 4 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: OPS does not allow bookmark links to H3 so I added an anchor tag.

Comment (DCtheGeek) [reply reference/5.1/Microsoft.PowerShell.Utility/Compare-Object.md]: I bookmark to H3s all the time... I don't think that's an accurate statement.

**Comment by sdwheeler** [comment]
@DCtheGeek #sign-off

**Pull request merged**